app/src/main/java/com/conupods/OutdoorMaps/MapInitializer.java

```diff
@@ -7,18 +7,154 @@
 
-    public MapInitializer(CameraController CameraController) {
-        mCameraController = CameraController;
+    public MapInitializer(CameraController cameraController, IndoorBuildingOverlays indoorBuildingOverlays, OutdoorBuildingOverlays outdoorBuildingOverlays, GoogleMap map) {
+        mCameraController = cameraController;
+        mIndoorBuildingOverlays = indoorBuildingOverlays;
+        mOutdoorBuildingOverlays = outdoorBuildingOverlays;
+        mMap = map;
+    }
+
+    public void onCameraChange(){
+
+        mMap.setOnCameraIdleListener(()->{
+
+            LatLngBounds bounds = mMap.getProjection().getVisibleRegion().latLngBounds;
+
+            if (mMap.getCameraPosition().zoom > 18.3) {
+                mOutdoorBuildingOverlays.removePolygons();
+                if (bounds.contains(centerOfJMSB)) {
+                    mIndoorBuildingOverlays.displayOverlayJMSB();
+                    mIndoorBuildingOverlays.showButtonsJMSB();
+                }
+                else if(bounds.contains(centerOfHall)) {
+                    mIndoorBuildingOverlays.displayOverlayHall();
+                    mIndoorBuildingOverlays.showButtonsHALL();
+                }
+                else if(bounds.contains(centerOfLOYCC)) {
+                    mIndoorBuildingOverlays.displayOverlayLOYCC();
+                }
+                else if(bounds.contains(centerOfLOYVL)) {
+                    mIndoorBuildingOverlays.displayOverlayLOYVL();
+                    mIndoorBuildingOverlays.showButtonsLOYVL();
+                }
+                else {
+                        mIndoorBuildingOverlays.hideLevelButton();
+                        mIndoorBuildingOverlays.removeOverlay();
+                }
+            } else {
+                    mIndoorBuildingOverlays.hideLevelButton();
+                    mIndoorBuildingOverlays.removeOverlay();
+                    mOutdoorBuildingOverlays.overlayPolygons();
+
+            }
+        });
+    }
+
+    private void changeButtonColors(List<Button> floorButtons){
+
+        for(Button button : floorButtons) {
+            if(!button.isPressed()){
+                button.setBackgroundColor(Color.WHITE);
+            }
+            if(button.isPressed()) {
+                button.setBackgroundColor(Color.LTGRAY);
+            }
+        }
+
+    }
+
+    public void initializeFloorButtons(View floorButtons) {
+        //Listener for floor buttons, display appropriate floor blueprint
+
+        //TODO logic for changing button colors on clickEvent
+        List<Button> buttonsHALL = new ArrayList<Button>();
+        List<Button> buttonsJMSB = new ArrayList<Button>();
+        List<Button> buttonsLOYCC = new ArrayList<Button>();
+
+
+        Button loy_vl1 = (Button) floorButtons.findViewById(R.id.loy_vl1);
+        loy_vl1.setOnClickListener((View v) -> {
+            mIndoorBuildingOverlays.changeOverlay(7, "VL");
+            changeButtonColors(buttonsLOYCC);
```

Comment:
this duplicate code should be turned into a private function where u pass the vars as parameters